### PR TITLE
feat(subgraph) - Add more details for receipts

### PR DESCRIPTION
**Pull request by kalidiagne**
# Description
We need `keyId` and `owner` for receipts to be able to get all receipts for a specific `keyId` of an owner 

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: subgraph/src/public-lock.ts [hunk 1]
```diff
@@ -239,7 +238,11 @@ export function handleKeyExtended(event: KeyExtendedEvent): void {
     key.save()
   }
   // create receipt
-  createReceipt(event)
+  if (key && key.owner) {
+    createReceipt(event, key.owner, event.params.tokenId)
+  } else {
+    createReceipt(event, Address.fromString(''), event.params.tokenId)
```
Comment (clemsos) [inline subgraph/src/public-lock.ts:244]: not sure when you end up with an empty key owner on renewal? Also this `Address.fromString('')` gives a null address?

File: subgraph/src/public-lock.ts [hunk 2]
```diff
@@ -269,7 +272,11 @@ export function handleRenewKeyPurchase(event: RenewKeyPurchaseEvent): void {
   }
 
   // create receipt
-  createReceipt(event)
+  if (key && key.owner) {
+    createReceipt(event, key.owner, tokenId.value)
+  } else {
+    createReceipt(event, Bytes.empty(), tokenId.value)
```
Comment (clemsos) [inline subgraph/src/public-lock.ts:278]: Same as above, when does a key without an owner happen in renewal? We should not pass empty Bytes but at least a null address

**Review by clemsos — COMMENTED** [review]
Seems weird to store an empty key owner in a receipt. People buy or renew a key for someone?

File: subgraph/schema.graphql [hunk 3]
```diff
@@ -110,4 +110,8 @@ type Receipt @entity {
   amountTransferred: BigInt
   "Total gas paid"
   gasTotal: BigInt!
+  "Refer to token id"
+  tokenId: BigInt
+  "Key owner of the receipt"
+  owner: String
```
Comment (julien51) [inline subgraph/schema.graphql:116]: That's a bad idea as a single receipt can apply to multiple keys for multiple owners.

**Comment by julien51** [comment]
Kali I removed the `owner` on purpose because there can be a single receipt for multiple okeys and owners. I don't understand why you need to add these values? Please clarify.

**Comment by kalidiagne** [comment]
> Kali I removed the `owner` on purpose because there can be a single receipt for multiple okeys and owners. I don't understand why you need to add these values? Please clarify.

to retrieve all the receipts by `owner +keyId` in receipt page details.
I can just use the hash of the transactions eventually.

**Review by julien51 — CHANGES_REQUESTED** [review]
Can we closer this please as this is not the right approach?

**Comment by kalidiagne** [comment]
yes, let's close it!

**Pull request closed without merging**